### ISSUE TITLE
Rewrite Connection class using Python's built-in 'abc' module

### ISSUE THREAD

**Issue by cher-nov** [LonamiWebs/Telethon#509]
$subj
Let's don't reinvent the OOP.
https://docs.python.org/3/library/abc.html

**Comment by Lonami** [comment]
If I understand correctly, the deal is making an abstract `Connection` and then using `ConnectionTcpFull`, `ConnectionAbridged`, instead manually picking up everything?

**Comment by cher-nov** [comment]
@Lonami, exactly. And let user specify the necessary connection class instead of `Enum` value.

**Comment by Lonami** [comment]
@cher-nov okay, but that will have to wait for another version ;)

**Comment by cher-nov** [comment]
@Lonami, no problem. Maybe I'll implement it later by myself.
